analyzers/src/SonarAnalyzer.Common/Extensions/SyntaxNodeAnalysisContextExtensions.cs

```diff
@@ -37,5 +38,16 @@ internal static class SyntaxNodeAnalysisContextExtensions
         /// <seealso href="https://github.com/dotnet/roslyn/issues/50989"/>
         internal static bool IsRedundantPositionalRecordContext(this SyntaxNodeAnalysisContext context) =>
             context.ContainingSymbol.Kind == SymbolKind.Method;
+
+        /// <summary>
+        /// Returns the <see cref="IMethodSymbol"/> of the entry point for an AzureFunction, if <code>Node</code> of <paramref name="context"/> is part
+        /// of the AzureFunction entry point.
+        /// </summary>
+        public static IMethodSymbol AzureFunctionMethod(this SyntaxNodeAnalysisContext context) =>
+            context.ContainingSymbol switch
+            {
+                IMethodSymbol method when method.HasAttribute(KnownType.Microsoft_Azure_WebJobs_FunctionNameAttribute) => method,
+                _ => null,
+            };
```

Comment:
This is abusing the switch expression, ternary operator should do the trick.
```suggestion
            context.ContainingSymbol is IMethodSymbol method && method.HasAttribute(KnownType.Microsoft_Azure_WebJobs_FunctionNameAttribute) => method
                ? method
                : null;
```